packages/gatsby/cache-dir/navigation.js

```diff
@@ -16,7 +16,13 @@ const redirectMap = redirects.reduce((map, redirect) => {
 }, {})
 
 function maybeRedirect(pathname) {
-  const redirect = redirectMap[pathname]
+  const redirect = Object.keys(redirectMap).find(redirect => {
+    if (redirectMap[redirect].ignoreCase) {
+      return redirect.toLowerCase() === pathname.toLowerCase()
+    }
+
+    return redirect === pathname
+  })
```

Comment:
This can get expensive. This is only used for browser redirects so we should be able to prepare this during build time.

```
let redirect = redirectMap[pathname];
if (!redirect) {
  redirect = redirectMap[pathname.toLowerCase()];
}
```